src/relay/backend/contrib/ethosn/codegen.cc

```diff
@@ -620,10 +626,10 @@ TVM_REGISTER_GLOBAL("relay.ethos-n.support.conv2d")
       auto err = EthosnAPI::QnnConv2d(call, &params);
       if (params.is_depthwise) {
         *rv = !err &&
-              m_Queries.IsDepthwiseConvolutionSupported(params.bias_info, params.weights_info,
+              m_Queries->IsDepthwiseConvolutionSupported(params.bias_info, params.weights_info,
```

Comment:
nit : might worth checking whether this is not null to ensure setup is called.

Reply:
It works because only a default variant was supported. Once another variant was selected, there was a mismatch between the variant that is used for the IsSupported functionality and the actual compilation. This patch prepares the way for that.
I restructured this a little to avoid having to call the setup from elsewhere; it is now a self-contained change.